src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -0,0 +1,50 @@
+#include "gcs_server.h"
+#include "default_job_info_access_handler.h"
+
+namespace ray {
+namespace gcs {
+
+GcsServer::GcsServer(const ray::gcs::GcsServerConfig &config)
+    : config_(config),
+      rpc_server_(config.server_name, config.server_port, config.server_thread_num) {}
+
+GcsServer::~GcsServer() { Stop(); }
+
+void GcsServer::Start() {
+  // Init redis gcs client
+  InitBackendClient();
+
+  // Register rpc service
+  job_info_access_handler_ = InitJobInfoAccessHandler();
+  job_info_access_service_.reset(
+      new rpc::JobInfoAccessGrpcService(main_service_, *job_info_access_handler_));
+  rpc_server_.RegisterService(*job_info_access_service_);
+
+  rpc_server_.Run();
+
+  boost::asio::io_context::work worker(main_service_);
```

Comment:
How about we don't use asio for the GCS service? We can just implement a normal async gRPC server like the core worker does (thanks to @edoakes ).

The benefit of this is that we can easily scale with multi-threading.

Reply:
> How about we don't use asio for the GCS service? We can just implement a normal async gRPC server like the core worker does (thanks to @edoakes ).
> 
> The benefit of this is that we can easily scale with multi-threading.

We already use GrpcServer inside gcs_server and it is easily to scale by the configuration of GcsServerConfig::server_thread_num. As we known, the redis client can not binds with grpc, so we still need asio::io_context to support redis client to communicate with redis asynchronously.